popup_renderer.c

```diff
@@ -260,11 +262,19 @@ int im_panel_init(struct wlpinyin_state *state) {
 	}
 
 	state->shm_pool =
-			wl_shm_create_pool(state->wl_shm, state->shm_pool_fd, DEFAULT_SHM_SIZE);
+		wl_shm_create_pool(state->wl_shm, state->shm_pool_fd, DEFAULT_SHM_SIZE);
 
-	state->popup_pango_ctx =
-			pango_font_map_create_context(pango_cairo_font_map_get_default());
+	PangoFontMap *font_map = pango_cairo_font_map_new();
+	if (font_map == NULL) {
+		wlpinyin_err("failed to create cairo font map");
+		return -1;
+	}
+	state->popup_pango_ctx = pango_font_map_create_context(font_map);
+	g_object_unref(font_map);
 	state->popup_pango_layout = pango_layout_new(state->popup_pango_ctx);
+	PangoFontDescription *desc = pango_font_description_from_string(POPUP_FONT);
+	pango_layout_set_font_description(state->popup_pango_layout, desc);
+	pango_font_description_free(desc);
```

Comment:
Setting font description does not need fontmap, just keep the old way of creating `pango_ctx`